flink-runtime/src/test/java/org/apache/flink/runtime/heartbeat/HeartbeatManagerTest.java

```diff
@@ -491,24 +508,24 @@ public void reportPayload(ResourceID resourceID, Object payload) {
 		}
 
 		@Override
-		public CompletableFuture<Integer> retrievePayload(ResourceID resourceID) {
+		public Integer retrievePayload(ResourceID resourceID) {
 			if (resourceID.equals(specialId)) {
-				return CompletableFuture.completedFuture(specialResponse);
+				return specialResponse;
 			} else {
-				return CompletableFuture.completedFuture(defaultResponse);
+				return defaultResponse;
 			}
 		}
 	}
 
-	static class TestingHeartbeatListener implements HeartbeatListener<Object, Object> {
+	static class SimpleTestingHeartbeatListener implements HeartbeatListener<Object, Object> {
```

Comment:
maybe de-duplicate by `extends TestingHeartbeatListener` and constructed by

```
super(
  resourceId -> future.complete(resourceID),
  (ignore1, ignore2) -> numberHeartbeatReports++,
  (ignore) -> CompletableFuture.completedFuture(payload)
);
this.payload = payload;
```

Reply:
This is not so easily possible because one cannot reference sub class fields when calling `super`.

Reply:
stupid mistake :P

Reply:
I removed this class by using the `TestingHeartbeatListener` directly.